### ISSUE TITLE
Pushing Updates to Templated Repos

### ISSUE THREAD

**Issue by brghena** [education/classroom#2054]
TLDR: What is the correct method for providing updates to templated repos?

We have been using github classroom for our student assignments with student code based off of a starter repo that they then make edits to. Not on each project, but at least a couple times a semester, we find bugs in the starter code, make a fix to our starter, and want to push the update out to students. For labs, we usually create a single lab assignment, and push additional folders to the lab starter repo each week with new code.

The general method for doing both of these has been to have students add our repo as a remote. Something along the lines of `git remote add staff https://github.com/61c-teach/su19-proj3-2-starter`. Then if there are updates, students can get them by running `git pull staff master`.

With the new templated repos, it seems that this doesn't work anymore. Adding the remote works fine, but when you try to pull, you get a `Warning: no common commits`.

What is the correct way of providing updates for templated repos? One option I found is to pull with the flag `--allow-unrelated-histories`?

It would be great if the methods for common actions like this were documented somewhere so we could know the common patterns most classes are using.

**Comment by gchallen** [comment]
We just got bitten by this same problem.

I really wish that this was explained more clearly at the place where the choice is made to use a templated repository. In our case we were always planning on release changes to an upstream repository, since we have students doing a multi-part assignment with new checkpoints released throughout the semester.

We're resorting to patch files, but that's a sad workaround given Git's capabilities in this area. Had I know that this was a consequence of using repository templates I would not have. (Of course, we've also been bitten by slow and failed clones in the past, and don't have that problem this semester.)
